src/me.rs

```diff
@@ -51,10 +52,18 @@ pub fn motion_estimation(
       let range = 32 as isize;
       let blk_w = bsize.width();
       let blk_h = bsize.height();
-      let x_lo = po.x - range;
-      let x_hi = po.x + range;
-      let y_lo = po.y - range;
-      let y_hi = po.y + range;
+      let border_w = 128 + blk_w as isize * 8;
+      let border_h = 128 + blk_h as isize * 8;
+      let cols = (rec.frame.planes[0].cfg.width + MI_SIZE - 1) / MI_SIZE;
```

Comment:
Perhaps the mutiplication and division by MI_SIZE could be replaced with shifts by MI_SIZE_LOG2.

Reply:
Compilers should be able to figure out that a division of an unsigned int by 4 (MI_SIZE) can be implemented with a shift.
I generally prefer using multiplications and divisions as there is no confusion about operator precedence